Library/Homebrew/test/cleanup_spec.rb

```diff
@@ -181,5 +181,21 @@
 
       expect(npm_cache).not_to exist
     end
+
+    it "cleans up files and directories with name containing -- if ARGV prune is all" do
+      a = (HOMEBREW_CACHE/"--a")
```

Comment:
The filename is meant to catch git (and other VCS) checkouts; for example, `~/Library/Caches/Homebrew/gist--git`. It'd be good to make the directory name more similar to the pattern we're trying to check for.